src/rust/engine/src/core.rs

```diff
@@ -88,6 +88,10 @@ impl hash::Hash for Key {
 }
 
 impl Key {
+  pub fn new_with_anon_type_id(id: Id) -> Key {
+    Key { id: id, type_id: TypeId(0) }
```

Comment:
Unless this id is reserved on the python side, this won't really be anonymous...